task-maker-cache/src/storage.rs

```diff
@@ -45,22 +38,21 @@ impl CacheFile {
         }
         let mut file = std::fs::File::open(&path)
             .with_context(|| format!("Cannot open cache file at {}", path.display()))?;
-        let mut magic = [0u8; MAGIC.len() + VERSION_MAX_LEN];
-        file.read_exact(&mut magic)
-            .context("Failed to read magic number")?;
-        if &magic[..MAGIC.len()] != MAGIC {
-            bail!(
-                "Cache magic mismatch:\nExpected: {:?}\nFound: {:?}",
-                MAGIC,
-                &magic[..MAGIC.len()]
-            );
-        }
-        if &magic[MAGIC.len()..MAGIC.len() + VERSION.len()] != VERSION.as_bytes() {
-            bail!(
+        let mut magic = [0u8; MAGIC.len()];
+
+        if file
+            .read_exact(&mut magic)
```

Comment:
Note to myself: this should work because the magic ends with a newline.

If the newline wasn't there this can break. For example, let's say there are 2 versions: `v0.1` and `v0.11`; running `v0.11` first, and then `v0.1`, without the newline the magic of the _old version_ is a prefix of the magic of the _new version_.

Probably it's better to write a comment on the `MAGIC` constant stating that a terminator should be placed at the end of `MAGIC` string for this reason.

Reply:
Added a comment